### ISSUE TITLE
Adding timestamp information to eopatch via ImportFromTiffTask function <= eo-learn version 0.10.2 [HELP]

### ISSUE THREAD

**Issue by torna4o** [sentinel-hub/eo-learn#403]
Hello everyone, 

**TL;DR: I want to add timestamp info to eopatch with ImportFromTiffTask in eo-learn v. 0.10.2, from a TIFF file I downloaded as .SAFE zip. How do I do that?**

I have been trying to run SentinelHub examples for eo-learn and eo-flow. I am utilizing ImportFromTiffTask from local.io and aside from a few missing things, I am able to run, for instance, land-cover-map with a single date and eopatch tiff. However, when I try to do the same with multi temporal files I would like to add timestamp as well. No matter what I do, however, it does not add itself. The metadata of entry TIFF file has extensive time data as it was loaded directly from .SAFE.zip file via gdal library.

I tried creating an "AddFeatureTask" as data: or timestamp: with FeatureType:TIMESTAMP, entries to no avail. From the log file I am getting this:

  data: {
    BANDS: numpy.ndarray(shape=(2, 500, 500, 10), dtype=float64)
    NDBI: numpy.ndarray(shape=(2, 500, 500, 1), dtype=float64)
    NDVI: numpy.ndarray(shape=(2, 500, 500, 1), dtype=float64)
    NDWI: numpy.ndarray(shape=(2, 500, 500, 1), dtype=float64)
  }
  mask: {
    CLM: numpy.ndarray(shape=(2, 500, 500, 1), dtype=uint8)
    IS_DATA: numpy.ndarray(shape=(2, 500, 500, 1), dtype=uint8)
    IS_VALID: numpy.ndarray(shape=(2, 500, 500, 1), dtype=bool)
  }
  scalar: {}
  label: {}
  vector: {}
  data_timeless: {}
  mask_timeless: {
    LULC: numpy.ndarray(shape=(500, 500, 1), dtype=uint8)
  }
  scalar_timeless: {}
  label_timeless: {}
  vector_timeless: {
    LULC_REFERENCE: geopandas.GeoDataFrame(columns=['RABA_PID', 'RABA_ID', 'VIR', 'AREA', 'STATUS', 'D_OD', 'lulcid', 'lulcname', 'geometry'], length=2812, crs=EPSG:32633)
  }
  meta_info: {}
  bbox: BBox(((503880.0, 5129220.0), (508880.0, 5134220.0)), crs=CRS('32633'))
  timestamp: []
)], **{'data': [datetime.datetime(2021, 6, 21, 9, 50, 9), datetime.datetime(2021, 7, 24, 10, 0, 29)]})
2022-04-07 12:30:00,893 root         DEBUG    EOWorkflow execution failed

As seen, instead of timestamp, it additionally put a 'data' type there for my manual datetime.datetime entries and it fails. 

hopelessly, I edited the ImportFromTiffTask as

add_data = ImportFromTiffTask(
    (FeatureType.DATA, 'BANDS'),
    (FeatureType.TIMESTAMP, 'timestamp'),
    timestamp_size=2,
    image_dtype=float,
    no_data_value= -99999
    )

but feels weird as it only has one dtype etc, and even if the code ran successfully with this, timestamp list was empty in eopatch features.

EOPatch(
  data: {
    BANDS: numpy.ndarray(shape=(2, 500, 500, 10), dtype=float64)
    NDBI: numpy.ndarray(shape=(2, 500, 500, 1), dtype=float64)
    NDVI: numpy.ndarray(shape=(2, 500, 500, 1), dtype=float64)
    NDWI: numpy.ndarray(shape=(2, 500, 500, 1), dtype=float64)
  }
  mask: {
    CLM: numpy.ndarray(shape=(2, 500, 500, 1), dtype=uint8)
    IS_DATA: numpy.ndarray(shape=(2, 500, 500, 1), dtype=uint8)
    IS_VALID: numpy.ndarray(shape=(2, 500, 500, 1), dtype=bool)
  }
  scalar: {}
  label: {}
  vector: {}
  data_timeless: {}
  mask_timeless: {
    LULC: numpy.ndarray(shape=(500, 500, 1), dtype=uint8)
  }
  scalar_timeless: {}
  label_timeless: {}
  vector_timeless: {
    LULC_REFERENCE: geopandas.GeoDataFrame(columns=['RABA_PID', 'RABA_ID', 'VIR', 'AREA', 'STATUS', 'D_OD', 'lulcid', 'lulcname', 'geometry'], length=2812, crs=EPSG:32633)
  }
  meta_info: {}
  bbox: BBox(((503880.0, 5129220.0), (508880.0, 5134220.0)), crs=CRS('32633'))
  timestamp: []
)

From what I get, SentinelHubInputTask uses timestamp for downloading from Sentinel Hub with charges, yet for learning purposes it is very prohibitive to be charged for each download. And especially eo-flow "examples" assumes we have all things like eopatches downloaded etc. , so even running the examples here rather than my own trials, I need to be able to create an identical eopatch to usual money charged version from Tiff I downloaded in other ways.

In case required, I am ready to share additional information or code, as said, on Google Colab and on Windows 10 with Python 3.7 and just direct pip installing the eo-learn was sufficient to run the land-cover-map algorithm with single tiff.

**Comment by batic** [comment]
Hi @torna4o 

the `timestamp` feature is just a list of `datetime` objects. 

So something like that should work:

```python
eopatch.timestamp = [ datetime.datetime(2022,3,18,12,0,0), datetime.datetime(2022,4,18,12,0,0)]
```
and would add two timestamps to the eopatch.

**Comment by torna4o** [comment]
Hello @batic 
This very simple way worked for me, too. I feel a bit ashamed but life is like that time to time.
After checking its output, I also managed to directly obtain Sentinel-2's .SAFE file metadata and pipe it to here as:
`eopatch.timestamp = [obj.GetMetadata()['GENERATION_TIME']]
`
where obj is _osgeo.gdal.Dataset_ retrieved after
`obj = gdal.Open(".SAFE zip file downloaded from copernicus sci-hub")`
Thank you very much.
